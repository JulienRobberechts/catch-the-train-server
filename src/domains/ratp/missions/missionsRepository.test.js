const { getSchedulesForMissions } = require("./missionsRepository");

const {
  getMissionDetailMocked,
} = require("../../../adapters/ratp-api/factory");

describe("missionsRepository", () => {
  describe("getSchedulesForMissions", () => {
    test("should import Missions Schedules", async () => {
      const missionCodes = ["NANI", "NOTE"];
      const missionsSchedules = await getSchedulesForMissions(
        getMissionDetailMocked,
        missionCodes
      );
      expect(missionsSchedules).toMatchSnapshot();
    });
  });
});
