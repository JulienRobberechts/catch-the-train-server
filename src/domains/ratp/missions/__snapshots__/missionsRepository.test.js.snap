// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missionsRepository getSchedulesForMissions should import Missions Schedules 1`] = `
Array [
  Object {
    "mission": "NANI",
    "stations": Array [
      "cergy+le+haut",
      "cergy+st+christophe",
      "cergy+prefecture",
      "neuville+universite",
      "conflans+fin+d'oise",
      "acheres+ville",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NOTE",
    "stations": Array [
      "poissy",
      "acheres+grand+cormier",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
]
`;
